org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Scanner.java

```diff
@@ -4861,12 +4860,12 @@ void updateScanContext(int token) {
 
 	@Override
 	public boolean isParsingModuleDeclaration() {
-		return true;
+		return Scanner.this.insideModuleInfo;
 	}
 
 	public ScanContext getScanContext(char[] src, int begin) {
 		this.scanner.setSource(src);
-		this.scanner.resetTo(0, begin);
+		this.scanner.resetTo(0, begin, isParsingModuleDeclaration());
 		goForCompilationUnit();
```

Comment:
Change the third parameter to `true` if you need the version with the third parameter please. Again at this place we are inside module-info

Reply:
understood. I'll rename the class to make it more obvious, OK?